tests/test_sequence.py

```diff
@@ -137,6 +138,288 @@ def test_magnetic_field():
     assert np.all(seq3_.magnetic_field == np.array((1.0, 0.0, 0.0)))
 
 
+def test_switch_device():
```

Comment:
What happened to splitting this into multiple functions? This test function is definitely too big now.

Reply:
You can make one function for testing all the errors and another for testing correct behaviour, for example. If you need the test devices in both, put them in one or more fixtures so you don't rewrite them